io/key/key.go

```diff
@@ -29,6 +29,7 @@ type InputOp struct {
 // It replaces any previous SoftKeyboardOp.
 type SoftKeyboardOp struct {
 	Show bool
+	Mode KeyboardMode
```

Comment:
What if an editor changes input mode but not focus? Input mode seems to belong to InputOp.

Reply:
The `KeyboardMode` only affects the on-screen-keyboard, which is only usefull when the `Show` is `true`. I actually think about replace the `Show: false` to `InvisibleKeyboard`. However, it would be a API change, so I avoid to change that.

----------

I don't know if it belongs to `InputOp` because it only affects the on-screen-keyboard. Who is responsable to show the virtual-keyboard is the `SoftKeyboardOp`. _That doesn't prevent the user to type anything on the physical keyboard._

Reply:
SoftKeyboardOp{Show:true} is not the only way to show the keyboard. The keyboard may appear on screen at all time, the Android Activity has a setting for displaying the keyboard when starting. Specifically, Android may call onCreateInputConnection at any time, but there may not have been a SoftKeyboardOp to update the InputMode.

Taking a step back, SoftKeyboardOp is an action op (show or hide the keyboard), whereas InputOp is a passive description of the widget. InputMode is a description of the widget, and therefore belongs in InputOp.

Reply:
I change it to `InputOp`. _If Android calls `onCreateInputConnection` at random, it will set the lastest mode._